tree_unique_args/src/interpreter.rs

```diff
@@ -50,6 +51,19 @@ pub fn typecheck(e: &Expr, arg_ty: &Option<Type>) -> Result<Type, TypeError> {
                 )),
             }
         }
+        Expr::Concat(tuple_1, tuple_2) => {
+            let ty_tuple_1 = typecheck(tuple_1, arg_ty)?;
+            let ty_tuple_2 = typecheck(tuple_2, arg_ty)?;
+            match (ty_tuple_1.clone(), ty_tuple_2) {
```

Comment:
Why do you only need a clone on the first?

Reply:
Good catch- we need to clone because we need the value later for the error message.
changed the code to clone both and more correct error messages